embassy-net/src/lib.rs

```diff
@@ -107,6 +116,8 @@ struct Inner<D: Driver> {
     config: Option<StaticConfig>,
     #[cfg(feature = "dhcpv4")]
     dhcp_socket: Option<SocketHandle>,
+    #[cfg(feature = "dns")]
+    dns_socket: Option<SocketHandle>,
```

Comment:
I think the Option could be removed. If DNS support is enabled, there should always be a socket.